code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -1976,42 +1976,57 @@
 
 	if(isnull(turflist))
 		return
+
 	var/turf/prev_turf
 	var/distance = 0
 
-	turf_loop:
-		for(var/turf/T in turflist)
-			distance++
+	for(var/turf/T in turflist)
```

Comment:
These should already come in `turf` format, so something like this would be more efficient, to avoid `istype` checks:
```
for(var/X in turflist)
	var/turf/T = X
```